### PR TITLE
gcc 7.1.0

### PR DESCRIPTION
**Pull request by fxcoudert**
Preparing for imminent GCC 7.1.0 release.

Removing some options, as java was dropped, `all-languages` is now identical to `jit` (and the name is really bad, cause it never built all languages),  `with-nls` and `without-fortran` are never used according to analytics.

### CODE REVIEW

**Comment by CarterLi** [comment]
> with-nls and without-fortran are never used according to analytics.

I'm using it 😸 

```
 $ gcc-6 -v
使用内建 specs。
COLLECT_GCC=gcc-6
COLLECT_LTO_WRAPPER=/usr/local/Cellar/gcc/6.3.0_1/libexec/gcc/x86_64-apple-darwin16.4.0/6.3.0/lto-wrapper
目标：x86_64-apple-darwin16.4.0
配置为：../configure --build=x86_64-apple-darwin16.4.0 --prefix=/usr/local/Cellar/gcc/6.3.0_1 --libdir=/usr/local/Cellar/gcc/6.3.0_1/lib/gcc/6 --enable-languages=c,c++,objc,obj-c++,fortran,jit --program-suffix=-6 --with-gmp=/usr/local/opt/gmp --with-mpfr=/usr/local/opt/mpfr --with-mpc=/usr/local/opt/libmpc --with-isl=/usr/local/opt/isl --with-system-zlib --enable-stage1-checking --enable-checking=release --enable-lto --with-build-config=bootstrap-debug --disable-werror --with-pkgversion='Homebrew GCC 6.3.0_1 --with-jit --with-nls --without-multilib' --with-bugurl=https://github.com/Homebrew/homebrew-core/issues --disable-multilib --enable-host-shared
线程模型：posix
gcc 版本 6.3.0 (Homebrew GCC 6.3.0_1 --with-jit --with-nls --without-multilib)
```

**Comment by petere** [comment]
I think `with-nls` should just be the default and not even be an option. With that kind of option, few people would go and dig it out and then rebuild the package. But many people would use it if it were enabled by default.

macOS is an internationalized operating system. Homebrew packages should just enable NLS and other l10n facilities if it's available and doesn't have any serious drawbacks.

In any case, it's perhaps not ideal this sneak this into an unrelated change?

**Comment by fxcoudert** [comment]
For me, the main problem is that GCC translations are, for the most part, very incomplete: https://translationproject.org/domain/gcc.html

I've readded the `--with-nls` option.

**Comment by RandomDSdevel** [comment]
Just wanted to note that I thumbs-upped @CarterLi's reply about `--with-nls` since I'm using it too, though, for me, it's just 'cause I like Unicode.  😀😜  (Should probably just have left it at the GitHub Reaction since nobody around here really likes 'Me, too!' posts, but I felt I had to say my piece.  Shutting up now…)

File: Formula/gcc@6.rb
Comment (JCount) [inline Formula/gcc@6.rb]: This block creates a conflict with the updated `gcc` if both are installed simultaneously.

**Review by JCount — CHANGES_REQUESTED** [review]

File: Formula/gcc@6.rb
Comment (fxcoudert) [inline Formula/gcc@6.rb]: Well spotted!

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Review by JCount — DISMISSED** [review]

File: Formula/gcc.rb
Comment (ilovezfs) [inline Formula/gcc.rb]: Since we have a working patch I would like to see this removed and replaced with the patch before this PR is merged.

**Review by ilovezfs — CHANGES_REQUESTED** [review]

File: Formula/gcc.rb
Comment (fxcoudert) [inline Formula/gcc.rb]: Yes I'm doing it :)

File: Formula/gcc.rb
Comment (ilovezfs) [inline Formula/gcc.rb]: It would be good to
1. find out if we can get it to use the superenv shim
2. put this patch in the formula-patches repository if we're going to have to carry it around for a while
3. create upstream issue about the trouble using the `--with-boot-ldflags` option successfully

**Review by ilovezfs — DISMISSED** [review]

File: Formula/gcc.rb
Comment (fxcoudert) [inline Formula/gcc.rb]: Yes I intend to do all that, but don't want to block this PR for now… if this passes, I want to commit it, get the new version available for users, and sort out the minor points later.

Comment (ilovezfs) [inline Formula/gcc.rb]: At minimum I'd like to see (2) taken care of. We are even talking about completely deprecating patch :DATA so it's :cry: to see a new one.

File: Formula/gcc.rb
Comment (ilovezfs) [inline Formula/gcc.rb]: Should be `https://raw.githubusercontent.com/Homebrew/formula-patches/32cf103/gcc/7.1.0-headerpad.patch`

File: formula_renames.json
Comment (ilovezfs) [inline formula_renames.json]: This should probably become `"gcc6": "gcc@6",`

**Review by ilovezfs — APPROVED** [review]

**Comment by zbeekman** [comment]
FYI, no need for `opencoarrays: revision for gcc` as of OpenCoarrays-1.8.10: https://github.com/Homebrew/homebrew-core/pull/13254

**Pull request closed without merging**
